lib/core-integration/src/Test/Integration/Scenario/API/Byron/Migrations.hs

```diff
@@ -115,66 +117,259 @@ spec :: forall n.
+                    error "Unable to compare plans."
+
+    describe "BYRON_MIGRATE_01 - \
+        \After a migration operation successfully completes, the correct \
+        \amounts eventually become available in the target wallet for an \
+        \arbitrary number of specified addresses, and the balance of the \
+        \source wallet is completely depleted."
+        $ do
+            testAddressCycling "Random" fixtureRandomWallet  1
+            testAddressCycling "Random" fixtureRandomWallet  3
+            testAddressCycling "Random" fixtureRandomWallet 10
+            testAddressCycling "Icarus" fixtureIcarusWallet  1
+            testAddressCycling "Icarus" fixtureIcarusWallet  3
+            testAddressCycling "Icarus" fixtureIcarusWallet 10
+
+    it "BYRON_MIGRATE_02 - \
+        \Can migrate a large wallet requiring more than one transaction."
+        $ \ctx -> runResourceT @IO $ do
+
+        -- NOTE:
+        --
+        -- Special mnemonic to which 200 legacy coins are attached in the
+        -- genesis file.
+        --
+        -- Out of these 200 coins:
+        --
+        --  - 100 coins are each worth 1 lovelace, and are expected to be
+        --    treated as dust.
+        --  - 100 coins are each worth 10,000,000,000 lovelace.
+        --
+        let sourceWalletMnemonic =
+                ["collect", "fold", "file", "clown"
+                , "injury", "sun", "brass", "diet"
+                , "exist", "spike", "behave", "clip"
+                ] :: [Text]
+        sourceWallet <- unsafeResponse <$> postByronWallet ctx
+            (Json [json|{
+                "name": "Big Byron Wallet",
+                "mnemonic_sentence": #{sourceWalletMnemonic},
+                "passphrase": #{fixturePassphrase},
+                "style": "random"
+            }|])
+        sourceBalance <- eventually "Source wallet balance is correct." $ do
+            response <- request @ApiByronWallet ctx
+                (Link.getWallet @'Byron sourceWallet) Default Empty
+            verify response
+                [ expectField (#balance . #available . #getQuantity)
+                    (`shouldBe` 1_000_000_000_100)
+                ]
+            return $ getFromResponse
+                (#balance . #available . #getQuantity) response
+
+        -- Create an empty target wallet:
+        targetWallet <- emptyWallet ctx
+        targetAddresses <- listAddresses @n ctx targetWallet
+        let targetAddressIds = targetAddresses <&>
+                (\(ApiTypes.ApiAddress addrId _ _) -> addrId)
+
+        -- Compute the expected migration plan:
+        responsePlan <- request @(ApiWalletMigrationPlan n) ctx
+            (Link.createMigrationPlan @'Byron sourceWallet) Default
+            (Json [json|{addresses: #{targetAddressIds}}|])
+        verify responsePlan
+            [ expectResponseCode HTTP.status202
+            , expectField
+                (#totalFee . #getQuantity)
+                (`shouldBe` 2_460_400)
```

Comment:
In integration tests, exact values like this may cause us pain in future.
It might be possible to make the test case more forgiving of changes, without reducing its effectiveness.

Reply:
Hi @rvl. I think I understand your concerns here.

However, I think there are good arguments in favour of using exact values:

- The new migration algorithm is completely deterministic: the plan is fully determined by the current UTxO, with no randomness. So if we start with the same fixture wallet and transaction constraints, we should always end up with precisely the same plan.

- To increase confidence that there are no flaky tests, I've run the entire migration integration test suite 500 times and confirmed that it always passes.

- Stating exact values means we can increase our confidence that the API is producing plans with reasonable fees, since fees are explicitly encoded as literals in the test code. I think it's good to be a bit paranoid here, as the potential cost to a user of miscalculating a fee is that they spend more ada than required (potentially without noticing).

- If someone alters the fixture wallet definitions or the fee calculation algorithm, then it will cause the tests to fail. However, they're likely to fail in an extremely obvious way, and it's likely that we can fix them very quickly just by entering new numbers. It also gives a fresh chance for a human pair of eyes to confirm that fee produced by the API is reasonable.